x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_suggestions.ts

```diff
@@ -120,15 +120,22 @@ function getBucketMappings(table: TableSuggestion, currentState?: State) {
   const currentXColumnIndex = prioritizedBuckets.findIndex(
     ({ columnId }) => columnId === currentLayer.xAccessor
   );
-  if (currentXColumnIndex) {
+  if (
```

Comment:
I found this pretty hard to parse. I suggest eliminating the double-negative by introducing a temporary variable:

```js
const currentXType = currentXColumnIndex > -1 && prioritizedBuckets[currentXColumnIndex].operation.dataType;

if (currentXType && currentXType !== 'date' && prioritizedBuckets[0].operation.dataType === 'date')

```

Reply:
You have a negation error in your sample there. This is what i came up with:
```ts
  const currentXDataType =
    currentXColumnIndex > -1 && prioritizedBuckets[currentXColumnIndex].operation.dataType;

  if (
    currentXDataType &&
    // make sure time gets mapped to x dimension even when changing current bucket/dimension mapping
    (currentXDataType === 'date' || prioritizedBuckets[0].operation.dataType !== 'date')
  ) {
```

Not sure whether that's easier to understand, though